sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/cloud/service/implementation/openai/OpenAIClientProperties.java

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.service.implementation.openai;
+
+import com.azure.ai.openai.OpenAIServiceVersion;
+import com.azure.spring.cloud.core.properties.AzureProperties;
+import com.azure.spring.cloud.core.provider.RetryOptionsProvider;
+import com.azure.spring.cloud.core.provider.authentication.KeyProvider;
+
+/**
+ * Configuration properties for Azure OpenAI.
+ */
+public interface OpenAIClientProperties extends AzureProperties, RetryOptionsProvider, KeyProvider {
+
+    /**
+     * Get the azure openai endpoint.
+     * @return the azure openai endpoint.
+     */
+    String getEndpoint();
+
+    /**
+     * Get the azure openai service version.
+     * @return the azure openai service version.
+     */
+    OpenAIServiceVersion getServiceVersion();
+
+    /**
+     * Get the NonAzureOpenAiKeyCredential used for public OpenAi authentication.
+     * @return the NonAzureOpenAiKeyCredential used for public OpenAi authentication.
+     */
+    String getNonAzureOpenAIKeyCredential();
```

Comment:
Shouldn't the name be getNonAzureOpenAIKey? Because this is a String type, instead of a credential class type.